roles/davinci_resolve/tasks/main.yml

```
@@ -108,6 +109,13 @@
     creates: "/tmp/{{ product | replace(' ', '_') }}_{{ pkgver }}.zip"
     follow_redirects: all
     status_code: 200
+  when: davinci_resolve_zip_file is undefined
+
+- name: Use pre-installed DaVinci Resolve
```

Please consider including the fix for #31 to this PR:
```yaml
- name: Validate existing DaVinci Resolve ZIP archive
  command: "unzip -t /tmp/{{ product | replace(' ', '_') }}_{{ pkgver }}.zip"
  register: zip_test
  failed_when: zip_test.rc != 0
  ignore_errors: true  # Ensure we proceed to deletion if this fails

- name: Remove invalid ZIP archive
  file:
    path: "/tmp/{{ product | replace(' ', '_') }}_{{ pkgver }}.zip"
    state: absent
  when: zip_test.failed  # Delete only if the validation failed
```
This should probably be modified to test also the file provided by the user, just in case is not valid.

---

Wouldn't it be a better way to just ignore_errors during the unzip task? `Unzip downloaded archive` so it will be removed if it is "unzippable"? :thinking: @leinardi

---

but with this approach that task will fail if and the user will have to figure out what's going on and start it again, manually, while, if we verify the zip file, before trying to unzip it and downloading it, it will be re-dowloaded automatically.

Just to be clear, I want to put this code before the download block, something like this:
```diff
Index: roles/davinci_resolve/tasks/main.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/roles/davinci_resolve/tasks/main.yml b/roles/davinci_resolve/tasks/main.yml
--- a/roles/davinci_resolve/tasks/main.yml	(revision 88d8cf079b24e631de538c3aa480911108c52ea3)
+++ b/roles/davinci_resolve/tasks/main.yml	(date 1741162777957)
@@ -100,6 +100,18 @@
     return_content: true
   register: register_response
 
+- name: Validate existing DaVinci Resolve ZIP archive
+  command: "unzip -t /tmp/{{ product | replace(' ', '_') }}_{{ pkgver }}.zip"
+  register: zip_test
+  failed_when: zip_test.rc != 0
+  ignore_errors: true  # Ensure we proceed to deletion if this fails
+
+- name: Remove invalid ZIP archive
+  file:
+    path: "/tmp/{{ product | replace(' ', '_') }}_{{ pkgver }}.zip"
+    state: absent
+  when: zip_test.failed  # Delete only if the validation failed
+
 - name: Download DaVinci Resolve from the provided URL
   ansible.builtin.uri:
     url: "{{ register_response.content | trim }}"
```
so that the Download block can re-download the file if the check failed

---

You are right in that sense. My idea was to ignore errors, and delete the `.zip` file (we already deleting the file after unzipping anyways). Therefore, if we have an error, we can just show it, ignore it, and finally delete the .zip file. In the second run, it will be fixed :thinking:  What do you think?